gateway/src/apicast/prometheus.lua

```diff
@@ -0,0 +1,23 @@
+local prometheus = require('nginx.prometheus')
+local assert = assert
+local dict = 'prometheus_metrics'
+
+if ngx.shared[dict] then
+  local init = prometheus.init(dict)
+
+  local metrics = { }
+  local __call = function(_, type, name, ...)
+    local metric_name = assert(name, 'missing metric name')
```

Comment:
Why do we check `name` and `type`? Can't we delegate this to the Prometheus lib?

Reply:
Because we actually use it. The type gets a function that is later called and the name is used as a key to store the metric. I had few tricky errors so decided to go for explicit ones.